### ISSUE TITLE
iPhone X: expose insets for safe region

### ISSUE THREAD

**Issue by cbracken** [flutter/flutter#12098]
## Steps to Reproduce

The iPhone X includes two screen regions that apps should avoid when displaying important content:
1. A notch in the display which, particularly in horizontal layouts, currently clips content in e.g., list views.
2. A thin bar along the bottom of the screen (bottom for both vertical and horizontal layouts) that indicates swipe up for app switcher/home, or swipe side to side to switch between apps.

Flutter should expose insets to allow framework users to identify these regions and work around them.

Example from Flutter Gallery:
![screen shot 2017-09-12 at 15 14 58](https://user-images.githubusercontent.com/351029/30454296-acffba34-9950-11e7-911a-e5b6971a056f.png)

See Apple's [Human Interface Guidelines for iPhone X](https://developer.apple.com/ios/human-interface-guidelines/overview/iphone-x)

## Flutter Doctor

```
[✓] Flutter (on Mac OS X 10.12.6 16G29, locale en-AU, channel master)
    • Flutter at /Users/cbracken/src/flutter/flutter
    • Framework revision dd7e313317 (51 minutes ago), 2017-09-14 12:28:21 -0700
    • Engine revision 57a1445a45
    • Tools Dart version 1.25.0-dev.11.0

[✓] Android toolchain - develop for Android devices (Android SDK 25.0.3)
    • Android SDK at /Users/cbracken/Library/Android/sdk
    • Platform android-25, build-tools 25.0.3
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_112-release-b06)

[✓] iOS toolchain - develop for iOS devices (Xcode 9.0)
    • Xcode at /Applications/Xcode9.0.app/Contents/Developer
    • Xcode 9.0, Build version 9A235
    • ios-deploy 1.9.2
    • CocoaPods version 1.3.1
```

**Comment by cbracken** [comment]
Fixed by flutter/engine#4302.

**Comment by cbracken** [comment]
Rolled into framework in flutter/flutter#12808.
